sklearn/tests/test_metaestimators_metadata_routing.py

```diff
@@ -92,38 +164,79 @@ def enable_slep006():
 - metaestimator: The metaestmator to be tested
 - estimator_name: The name of the argument for the sub-estimator
 - estimator: The sub-estimator
+- init_args: The arguments to be passed to the metaestimator's constructor
 - X: X-data to fit and predict
 - y: y-data to fit
-- routing_methods: list of all methods to check for routing
-- preserves_metadata: Whether the metaestimator passes the metadata to the
-  sub-estimator without modification or not. If it does, we check that the
-  values are identical. If it doesn't, no check is performed. TODO Maybe
-  something smarter could be done if the data is modified.
-
+- estimator_routing_methods: list of all methods to check for routing metadata
+  to the sub-estimator
+- preserves_metadata:
+    - True (default): the metaestimator passes the metadata to the
+      sub-estimator without modification. We check that the values recorded by
+      the sub-estimator are identical to what we've passed to the
+      metaestimator.
+    - False: no check is performed regarding values, we only check that a
+      metadata with the expected names/keys are passed.
+    - "subset": we check that the recorded metadata by the sub-estimator is a
+      subset of what is passed to the metaestimator.
```

Comment:
Can you please explain, @adrinjalali ? In which case would we test each case?

Reply:
for instance, `GridSearchCV` can take `sample_weight`, but select a subset of that and pass it to the sub-estimator. Other meta-estimators would forward the metadata as is.

Reply:
I have seen that `GridSearchCV`  is checked for passing a subset of the metadata to sub-estimators. My assumption is, because it uses an internal `splitter`? But why then is `LogisticRegressionCV` checked for the whole metadata to be passed (it also splits the data)?
My apologies, but I need more guidance here.